freemarker/src/main/java/org/jdbi/v3/freemarker/FreemarkerSqlLocator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.freemarker;
+
+import java.io.File;
+import java.io.FileReader;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import freemarker.cache.ClassTemplateLoader;
+import freemarker.cache.MultiTemplateLoader;
+import freemarker.cache.TemplateLoader;
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import net.jodah.expiringmap.ExpirationPolicy;
+import net.jodah.expiringmap.ExpiringMap;
+
+/**
+ * Locates SQL in <code>.sql.ftl/code> Freemarker files on the classpath.
+ */
+public class FreemarkerSqlLocator {
+    private static final Map<String, Template> CACHE = ExpiringMap.builder()
+            .expiration(10, TimeUnit.MINUTES)
+            .expirationPolicy(ExpirationPolicy.ACCESSED)
+            .build();
+
+    private FreemarkerSqlLocator() {}
+
+    private static File findTemplateDirectory(Class<?> type) {
+        try {
+            String classFolder = getPath(type);
+            URL resource = type.getClassLoader().getResource(classFolder);
+            if (resource != null) {
+                return new File(resource.toURI());
+            }
+        } catch (URISyntaxException ignored) { }
+        return null;
+    }
+
+    public static Template findTemplateOrFail(Class<?> type, String templateName) {
```

Comment:
I feel like the `OrFail` in the name is redundant.

If the template isn't there, the caller should expect this to bomb.